### PR TITLE
[ARCTIC-1537] support to manage optimizer groups in dashboard

### LINKED ISSUE #1537: [Feature]: support to manage optimizer groups in dashboard
### Description

1. Right now it is only possible to add the optimizer group configuration in config.yaml,  which is very inconvenient. We need to add functionality on the front end to support users to add it in the interface.
2. It is necessary to add a forced/required property of default optimizer group in catalog config, to make sure every table wherever in internal catalog or external catalog could have a default optimizer group and make TableRuntime properly initialized.

### Use case/motivation

1. facilitate users to manage optimizer groups in the dashboard for an improved management experience
2. make a forced relationship between catalog and resource group

### Describe the solution

1、add a `groups` tab under Optimizing module in dashboard, with a group list and creating group function.
image

2. add a required property under catalog config page.

### Subtasks

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### LINKED ISSUE #1544: [Subtask]: Introdue the concept of the InternalCatalog and ExternalCatalog
### Search before asking

- [X] I have searched in the [issues](https://github.com/NetEase/arctic/issues?q=is%3Aissue) and found no similar issues.

### Description

image

When creating a catalog:

1.  Choosing catalog type from `InternalCatalog` or `ExternalCatalog`
2.  When choosing an internal catalog, we should not choose the metastore type.
3.  When choosing an internal catalog, we support iceberg or mixed-iceberg to choose
4.  When choosing an external catalog, we must choose the metastoe type(Hive, Hadoop, Custom)
5.  When choosing an external catalog, we can choose the format (Iceberg and Mixed-Iceberg) or (Mixed-Hive) if the metastore type is Hive 

### Parent issue

https://github.com/NetEase/arctic/issues/1061

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by hameizi**
## Why are the changes needed?
resolve #1537, resolve #1544

## Brief change log

Add controller interfaces for managing resource groups.

## How was this patch tested?
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] Run test locally before making a pull request

## Documentation

  - Does this pull request introduces a new feature? (yes)
  - If yes, how is the feature documented? (docs)

### CODE REVIEW

**Review by Aireed — APPROVED** [review]
LGTM

File: ams/server/src/main/resources/mysql/ams-mysql-init.sql [hunk 1]
```diff
@@ -54,7 +54,6 @@ CREATE TABLE `resource`
     `total_memory`              bigint(30) DEFAULT NULL COMMENT 'optimizer use memory size',
     `start_time`                timestamp not null default CURRENT_TIMESTAMP COMMENT 'optimizer start time',
     `properties`                mediumtext COMMENT 'optimizer instance properties',
-    PRIMARY KEY (`resource_id`),
     KEY  `resource_group` (`group_name`)
```
Comment (shidayang) [inline ams/server/src/main/resources/mysql/ams-mysql-init.sql:57]: Perhaps we don't need the "resource_id" anymore?

File: ams/server/src/main/java/com/netease/arctic/server/dashboard/controller/OptimizerController.java [hunk 2]
```diff
@@ -207,5 +211,112 @@ public void scaleOutOptimizer(Context ctx) {
...
+   * url = /optimize/resourceGroups/create
+   */
+  public void createResourceGroup(Context ctx) {
+    Map<String, Object> map = ctx.bodyAsClass(Map.class);
+    String name = (String) map.get("name");
+    String container = (String) map.get("container");
+    Map<String, String> properties = (Map) map.get("properties");
+    ResourceGroup.Builder builder = new ResourceGroup.Builder(name, container);
+    builder.addProperties(properties);
+    optimizerManager.createResourceGroup(builder.build());
```
Comment (shidayang) [inline ams/server/src/main/java/com/netease/arctic/server/dashboard/controller/OptimizerController.java:245]: We need to translate the primary key conflict exception error message into user-friendly prompts, such as "Group already exists."

File: ams/server/src/main/java/com/netease/arctic/server/DefaultOptimizingService.java [hunk 3]
```diff
@@ -208,6 +209,25 @@ public void deleteOptimizer(String group, String resourceId) {
...
+  public void createResourceGroup(ResourceGroup resourceGroup) {
+    super.createResourceGroup(resourceGroup);
+    String groupName = resourceGroup.getName();
+    OptimizingQueue optimizingQueue = new OptimizingQueue(tableManager,
+        resourceGroup,
+        new ArrayList<>(),
+        new ArrayList<>(),
+        optimizerTouchTimeout,
+        taskAckTimeout);
+    optimizingQueueByGroup.put(groupName, optimizingQueue);
```
Comment (shidayang) [inline ams/server/src/main/java/com/netease/arctic/server/DefaultOptimizingService.java:222]: All operations need to be within a transaction.

File: ams/server/src/main/java/com/netease/arctic/server/DefaultOptimizingService.java [hunk 4]
```diff
@@ -208,6 +209,25 @@ public void deleteOptimizer(String group, String resourceId) {
...
+  }
+
+  @Override
+  public void deleteResourceGroup(String groupName) {
+    optimizingQueueByGroup.remove(groupName);
+    super.deleteResourceGroup(groupName);
```
Comment (shidayang) [inline ams/server/src/main/java/com/netease/arctic/server/DefaultOptimizingService.java:228]: All operations need to be within a transaction.

File: ams/server/src/main/resources/mysql/ams-mysql-init.sql (hunk 1 above)
Comment (hameizi) [reply ams/server/src/main/resources/mysql/ams-mysql-init.sql:57]: We need the resource_id to associate optimizer and actual running application information.

**Comment by wangtaohz** [comment]
The title should be `[ARCTIC-1537] xxxx` to match the community guide. @hameizi

File: ams/server/src/main/java/com/netease/arctic/server/DefaultOptimizingService.java [hunk 5]
```diff
@@ -208,6 +211,42 @@ public void deleteOptimizer(String group, String resourceId) {
...
+              optimizerTouchTimeout,
+              taskAckTimeout);
+          optimizingQueueByGroup.put(groupName, optimizingQueue);
+        })
+    );
+  }
+
+  @Override
+  public void deleteResourceGroup(String groupName) {
+    invokeConsisitency(() -> {
```
[the remark this answers is not in the document]
Comment (hameizi) [reply ams/server/src/main/java/com/netease/arctic/server/DefaultOptimizingService.java:234]: `super.deleteResourceGroup`  performs validation before deleting the group.

**Review by zhoujinsong — APPROVED** [review]

**Review by zhoujinsong — APPROVED** [review]

**Comment by XBaith** [comment]
Thanks, this PR is really helpfull

**Pull request merged**